package leetcode.problems.intersection_of_two_arrays;
//Given two integer arrays nums1 and nums2, return an array of their
//intersection
//. Each element in the result must be unique and you may return the result in any order.
//
//
//
//Example 1:
//
//Input: nums1 = [1,2,2,1], nums2 = [2,2]
//Output: [2]
//Example 2:
//
//Input: nums1 = [4,9,5], nums2 = [9,4,9,8,4]
//Output: [9,4]
//Explanation: [4,9] is also accepted.
//
//
//Constraints:
//
//1 <= nums1.length, nums2.length <= 1000
//0 <= nums1[i], nums2[i] <= 1000

import java.util.ArrayList;
import java.util.List;

class Solution {
	public int[] intersection(int[] nums1, int[] nums2) {
		int[] map = new int[1001];
		for (int num : nums1) {
			map[num] = 1;
		}
		List<Integer> nums = new ArrayList<>();
		for (int num : nums2) {
			if (map[num] == 1) {
                map[num]++;
				nums.add(num);
			}
		}
		int[] result = new int[nums.size()];
        for (int i = 0; i < nums.size(); i++) {
            result[i]=nums.get(i);
        }
        return result;
	}
}